### PR TITLE
.gitignore and dev_depedency updates

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,5 @@
 # Files and directories created by pub
-.buildlog
+.dart_tool
 .packages
-.project
 .pub
```
Comment (natebosch) [inline .gitignore]: this isn't created anymore is it?

File: .gitignore [hunk 2]
```diff
@@ -1,14 +1,5 @@
 # Files and directories created by pub
-.buildlog
+.dart_tool
 .packages
-.project
```
Comment (natebosch) [inline .gitignore]: are some of these from IntelliJ or something? Do we want to remove them?

File: .gitignore [hunk 3]
```diff
@@ -1,14 +1,5 @@
...
 .pub
-**/build
-**/packages
-
-# Directory created by dartdoc
-doc/api/
```
Comment (natebosch) [inline .gitignore]: why remove this?

File: pubspec.yaml [hunk 4]
```diff
@@ -15,15 +15,8 @@ dependencies:
 
 dev_dependencies:
   build: ^0.12.0
-  build_runner: ^0.7.9
+  build_runner: '>=0.7.9 <0.9.0'
```
Comment (natebosch) [inline pubspec.yaml]: it's a dev dependency and we're already restricted to dart 2 - why do we need to support old build_runner versions?

File: .gitignore (hunk 1 above)
Comment (kevmoo) [reply .gitignore]: It is a VERY old thing from Dart Editor days.

File: .gitignore (hunk 3 above)
Comment (kevmoo) [reply .gitignore]: packages dir is no longer created
We don't generate docs in this package

File: pubspec.yaml (hunk 4 above)
Comment (kevmoo) [reply pubspec.yaml]: Was trying to just support latest. No issues bumping to `^0.8.0`

**Pull request merged**